src/vscode-dts/vscode.proposed.chatParticipantAdditions.d.ts

```diff
@@ -166,6 +174,10 @@ declare module 'vscode' {
 
 		textEdit(target: Uri, isDone: true): void;
 
+		notebookEdit(target: Uri, edits: NotebookEdit | NotebookEdit[]): void;
+
+		notebookEdit(target: Uri, isDone: true): void;
```

Comment:
Left as discussion point. I do not think we need the `isDone` overload.

Then again, thats the right way to signal start and end of notebooks.
Today we use `textEdit` with empty edit array and `true` to signal start and end.
Should we do the same for `notebookEdit` as well, thats what I've done, however looking at the API its not clear that is how it needs to be done.

Reply:
> Today we use textEdit with empty edit array and true to signal start and end.

You should be using `notebookEdit` instead. The start and end signals are important to quickly show/hide the in-place progress UIs (the chat pills and overlay widget) and we should be honest about the file that's being targeted, e.g notebook A vs text B vs notebook C etc